lib/compiler.js

```diff
@@ -480,18 +480,20 @@ function resolveMiddlewareParams(rootDir, params) {
 }
 
 function buildComponentInstructions(rootDir, componentConfig) {
-  return Object.keys(componentConfig).map(function(name) {
-    var sourceFile;
-    if (name.indexOf('./') === 0 || name.indexOf('../') === 0) {
-      // Relative path
-      sourceFile = path.resolve(rootDir, name);
-    } else {
-      sourceFile = require.resolve(name);
-    }
+  return Object.keys(componentConfig)
+    .filter(function(name) { return !!componentConfig[name]; })
+    .map(function(name) {
+      var sourceFile;
+      if (name.indexOf('./') === 0 || name.indexOf('../') === 0) {
+        // Relative path
+        sourceFile = path.resolve(rootDir, name);
+      } else {
+        sourceFile = require.resolve(name);
+      }
 
-    return {
-      sourceFile: sourceFile,
-      config: componentConfig[name]
-    };
-  });
+      return {
+        sourceFile: sourceFile,
+        config: componentConfig[name]
+      };
+    });
```

Comment:
Are you using tab characters here? Please ensure you use only spaces for indentation (you may need to change the settings in your editor).